factory/runType/runTypeComponentsHandler.go

```diff
@@ -265,6 +269,18 @@ func (mrc *managedRunTypeComponents) ValidatorStatisticsProcessorCreator() peer.
 	return mrc.runTypeComponents.validatorStatisticsProcessorCreator
 }
 
+// SCProcessorCreator returns the smart contract processor creator
+func (mrc *managedRunTypeComponents) SCProcessorCreator() scrCommon.SCProcessorCreator {
+	mrc.mutStateComponents.RLock()
+	defer mrc.mutStateComponents.RUnlock()
+
+	if mrc.runTypeComponents == nil {
```

Comment:
checl.ifnil ?

Reply:
on every if statement which looks like that.

Reply:
fixed